PlayCover/Views/Sidebar Views/AppFolderView.swift

```diff
@@ -0,0 +1,272 @@
+                        apps.apps.contains($0.info.bundleIdentifier)
+                    },
+                                      selectedBackgroundColor: $selectedBackgroundColor,
+                                      selectedTextColor: $selectedTextColor,
+                                      selected: $selected,
+                                      isList: $isList,
+                                      gridLayout: gridLayout)
+                }
+                .onTapGesture {
+                    selected = nil
+                }
+            } else {
+                VStack {
+                    Text("playapp.noSources.title")
+                        .font(.title)
+                        .padding(.bottom, 2)
+                    Text("playapp.noSources.subtitle")
+                        .font(.subheadline)
+                        .foregroundColor(.secondary)
+                    Button("playapp.importIPA") {
+                        if installVM.inProgress {
+                            Log.shared.error(PlayCoverError.waitInstallation)
+                        } else if downloadVM.inProgress {
+                            Log.shared.error(PlayCoverError.waitDownload)
+                        } else {
+                            selectFile()
+                        }
+                    }
+                }
+                .frame(maxWidth: .infinity, maxHeight: .infinity)
+            }
+            Button(NSLocalizedString("button.edit.folder", comment: ""), action: {
+                addSheetApps.toggle()
+            })
+            .padding()
+        }
+        .sheet(isPresented: $addSheetApps) {
+            var dynamicHeight: CGFloat {
+                let count = CGFloat(appsVM.apps.count) * 85
+                return min(count, 600)
+            }
+            VStack {
+                HStack {
+                    TextField(text: $appsEdited.name, label: {Text("folder.textfield.name")})
+                        .frame(height: 40)
+                    Picker(selection: $appsEdited.icon, label: Text("Icon")) {
+                        ForEach(icons, id: \.self) { icon in
+                            Image(systemName: icon)
+                        }
+                    }.fixedSize()
+                }
+                List(AppsVM.shared.apps, id: \.url) { app in
+                    AddAppSheet(addSheetApps: addSheetApps,
+                                isAppEnabled: apps.apps.contains(app.info.bundleIdentifier),
+                                app: app,
+                                appList: $appsEdited
+                    )
+                }
+                Spacer()
+                    .frame(height: 40)
+                HStack {
+                    Spacer()
+                    Button(NSLocalizedString("button.OK", comment: ""), action: {
+                        addSheetApps.toggle()
+                        apps.apps = appsEdited.apps
```

Comment:
I think it would be better to have these sets to be above the sheet toggle to ensure that the values are initialized when loaded out of the sheet.